### ISSUE TITLE
[REQUEST] DiscreteBCQ imitator loss

### ISSUE THREAD

**Issue by N-AtR** [takuseno/d3rlpy#110]
I have an imbalanced dataset and I want to use DiscreteBCQ for training an agent. My problem is that the imitator performance is not good because the dataset is imbalanced and the classification algorithm fails to predict action probabilities correctly! When I train the imitator considering weight of classes in loss function its performance is good. So in summary the main problem is that I cannot add weight to imitator loss in DiscreteBCQ algorithm. Is it possible to add this feature in the library?

Thanks

**Comment by takuseno** [comment]
@N-AtR Hi, thanks for the issue. Could you explain more about the `weight`? The algorithms are basically designed to reproduce the papers. Therefore, if that feature needs big changes, I might create a new algorithm.

**Comment by N-AtR** [comment]
Dear @takuseno Thank you. Sure! The imitator of the paper for discrete actions is a classifier that is optimized based on negative log likelihood loss. In pytorch this loss function gets weights for imbalanced datasets as you can see [here](https://pytorch.org/docs/stable/generated/torch.nn.NLLLoss.html). My question is that is it possible to pass appropriate weights for training imitator to DiscreteBCQ?
Please let me know if you have any other questions. Again thank you for quick response
